### PR TITLE
fix: Input affix use inline-flex

### LINKED ISSUE #20831: Input prefix suffix style
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://next.ant.design/components/input-cn/#components-input-demo-presuffix](https://next.ant.design/components/input-cn/#components-input-demo-presuffix)

### Steps to reproduce
Enter large amount of text.Prefix two texts have the same problem.
image

### What is expected?
 normal style

### What is actually happening?
error style 

| Environment | Info |
|---|---|
| antd | 4.0.0-rc.0 |
| React | latest |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

fix #20831

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Fix Input suffix will override with input content.      |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

-----
[View rendered components/input/demo/presuffix.md](https://github.com/ant-design/ant-design/blob/input-suffix/components/input/demo/presuffix.md)

### CODE REVIEW

**Comment by zombieJ** [comment]
改用 flex 布局，看了一圈应该没什么问题。帮忙再次确认一下。 @afc163

File: components/input/ClearableLabeledInput.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ interface BasicProps {
   style?: object;
   disabled?: boolean;
   direction?: any;
+  focused?: boolean;
```
Comment (afc163) [inline components/input/ClearableLabeledInput.tsx]: Props?

Comment (zombieJ) [reply components/input/ClearableLabeledInput.tsx]: Yes.  This PR #19310 create an additional `ClearableLabeledInput` component for clearable but render input element is in `Input`. `focused` state is get from input and pass to `ClearableLabeledInput`.

File: components/input/Input.tsx [hunk 2]
```diff
@@ -194,6 +212,8 @@ class Input extends React.Component<InputProps, InputState> {
       <input
         {...otherProps}
         onChange={this.handleChange}
+        onFocus={this.onFocus}
+        onBlur={this.onBlur}
```
Comment (afc163) [inline components/input/Input.tsx]: 没看出来 focused 放到上层的必要，直接 ClearableLabeledInput 里面消费掉如何？

Comment (afc163) [reply components/input/Input.tsx]: 不过放里面好像也挺麻烦。

Comment (zombieJ) [reply components/input/Input.tsx]: 考虑是 ClearableLabeledInput 内部用的 `cloneElement` 来加额外的属性：
```jsx
{React.cloneElement(labeledElement, { style: null })}
```

如果要加 `onFocus` 、`onBlur` 就需要变成：
```jsx
const passProps = {
  style: null,
};

if (React.isValidateElement(element)) {
  const { onFocus, onBlur } = element.props;
  passProps.onFocus = (e) => {
    // ...
  };
  passProps.onBlur = (e) => {
    // ...
  };
}

...
```

看起来会比较恶心，不如和原本处理事件逻辑 `onChange`、`onKeyDown` 一样放在外面。

**Comment by afc163** [comment]
没预览可看了，codesandbox ci 打不开。

https://codesandbox.io/s/antd-reproduction-template-7ql18

**Comment by zombieJ** [comment]
试试 gitpod: https://gitpod.io/#https://github.com/ant-design/ant-design/pull/20872

**Pull request merged**